pkg/integrations/cadvisor/cadvisor.go

```diff
@@ -0,0 +1,166 @@
+var disabledMetricsSet container.MetricSet = container.MetricSet{
+	container.MemoryNumaMetrics:              struct{}{},
+	container.NetworkTcpUsageMetrics:         struct{}{},
+	container.NetworkUdpUsageMetrics:         struct{}{},
+	container.NetworkAdvancedTcpUsageMetrics: struct{}{},
+	container.ProcessSchedulerMetrics:        struct{}{},
+	container.ProcessMetrics:                 struct{}{},
+	container.HugetlbUsageMetrics:            struct{}{},
+	container.ReferencedMemoryMetrics:        struct{}{},
+	container.CPUTopologyMetrics:             struct{}{},
+	container.ResctrlMetrics:                 struct{}{},
+	container.CPUSetMetrics:                  struct{}{},
+}
+
+// GetIncludedMetrics applies some logic to determine the final set of metrics to be scraped and returned by the cAdvisor integration
+func (c *Config) GetIncludedMetrics() (container.MetricSet, error) {
+	var enabledMetrics, includeMetrics container.MetricSet
+	// Clear default disabled metrics if explicit disabled metrics are configured
+	if len(c.DisabledMetrics) > 0 {
+		disabledMetricsSet = container.MetricSet{}
+	}
+	for _, d := range c.DisabledMetrics {
+		if err := disabledMetricsSet.Set(d); err != nil {
+			return includeMetrics, fmt.Errorf("failed to set disabled metric: %w", err)
+		}
+	}
+
+	for _, e := range c.EnabledMetrics {
+		if err := enabledMetrics.Set(e); err != nil {
+			return includeMetrics, fmt.Errorf("failed to set enabled metric: %w", err)
+		}
+	}
+
+	if len(enabledMetrics) > 0 {
+		includeMetrics = enabledMetrics
+	} else {
+		includeMetrics = container.AllMetrics.Difference(disabledMetricsSet)
+	}
+
+	return includeMetrics, nil
+}
+
+// NewIntegration creates a new cadvisor integration
+func (c *Config) NewIntegration(logger log.Logger) (integrations.Integration, error) {
+	return New(logger, c)
+}
+
+// Integration implements the cadvisor integration
+type Integration struct {
+	c *Config
+	i *integrations.CollectorIntegration
+}
+
+// Run holds all the configuration logic for globals, as well as starting the resource manager and registering the collectors with the collector integration
+func (i *Integration) Run(ctx context.Context) error {
+	// Do gross global configs. This works, so long as there is only one instance of the cAdvisor integration
+	// per host.
+	// Containerd
+	containerd.ArgContainerdEndpoint = &i.c.Containerd
+	containerd.ArgContainerdNamespace = &i.c.ContainerdNamespace
+
+	// Docker
+	docker.ArgDockerEndpoint = &i.c.Docker
+	docker.ArgDockerTLS = &i.c.DockerTLS
+	docker.ArgDockerCert = &i.c.DockerTLSCert
+	docker.ArgDockerKey = &i.c.DockerTLSKey
+	docker.ArgDockerCA = &i.c.DockerTLSCA
+
+	// Raw
+	raw.DockerOnly = &i.c.DockerOnly
+
+	// Only using in-memory storage, with no backup storage for cadvisor stats
+	memoryStorage := memory.New(i.c.StorageDuration, []storage.StorageDriver{})
+
+	sysFs := sysfs.NewRealSysFs()
+
+	collectorHTTPClient := http.Client{}
```

Comment:
nit: use `var` for zero value declarations: 

```go
var collectorHTTPClient http.Client
```

(From [uber-go/style](https://github.com/uber-go/guide/blob/master/style.md))